src/core/core.controller.js

```diff
@@ -537,11 +539,14 @@ class Chart {
 				meta.controller.updateIndex(i);
 				meta.controller.linkScales();
 			} else {
-				const ControllerClass = controllers[meta.type];
-				if (ControllerClass === undefined) {
-					throw new Error('"' + meta.type + '" is not a chart type.');
-				}
-
+				const controllerDefaults = defaults[type];
+				const ControllerClass = registry.getController(type);
+				Object.assign(ControllerClass.prototype, {
```

Comment:
why can't we still set these in the controllers?

Reply:
So we don't pull the element types with the controller. So when you derive from a controller, you can change the element types and the original ones can be shaked.
This also enabled changing the element type only, for example: 
```js
Chart.register(MyBar);
Chart.defaults.bar.dataElementType = 'MyBar'`;
```